pages/docs/concepts/asyncapi-document/payload-schema.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Payload Schema
+weight: 239
+---
+
+The payload schema defines a message's format, data types, and properties to ensure that the payload follows a specific structure and data format.
+
+In the AsyncAPI specification, you can use schema languages such as JSON or Avro Schema to create the payload schema. Enabling consumers to gain insights into the structure and data types of the payload.
+
+You should use Avro schemas to handle complex messages and structured data models. Avro schemas support complex types like arrays and records. You can create Avro schemas by writing the JSON representation or using tools like Avro4s UI to generate the schema from valid JSON.
+
+Use npm or yarn to install the `@asyncapi/avro-schema-parser` package to enable Avro schema parsing in AsyncAPI.
+
+## Define Avro schema
+
+Define the Avro schema for the message payload. You can choose one of the following methods,
+
+- Embedded notation: Define the Avro schema within the message payload property.
+- Remote reference: Specify the schema using an absolute remote endpoint, such as `$ref: 'https://schemas.example.com/user'`.
+- Local reference: Specify the schema using a relative reference, such as `$ref: './user-signedup.avsc#/User'`.
+
+The diagram below defines the AsyncAPI specification file using the local reference method.
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+style E fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[Define Avro Schema]
+B[Local Reference]
+C[Avro Schema File]
+D[Message Payload]
+E[AsyncAPI Specification File]
+
+A -->|One of the following methods:| B
+B -->|Specify schema using relative reference| D
+E -->|Uses local reference method| C
+C-->|Defines structure of message payload| D
+```
```

Comment:
Not sure I understand the purpose of this diagram 🤔

Reply:
As mentioned in another PRs, I've explained to Fran in our Spec 3 Docs meeting why we are adding diagrams in our project scope. Fran should now understand why we have several diagrams.

Reply:
I understand we need diagrams, but not sure what this diagram represents. 

> The diagram below defines the AsyncAPI specification file using the local reference method.

I don't think diagram `defines the AsyncAPI specification file using the local reference method`

Reply:
Yeah, I find it confusing. Still not sure of the meaning of this diagram, even after you explained it @alequetzalli.

Reply:
@derberg @fmvilas , i updated the diagram. Could you help me understand how i can make this image better?

Reply:
so diagram description says:
`The diagram below defines the AsyncAPI specification file using the local reference method.`

I find it confusing as I cannot imagine how diagram can define something, and how `local` is special in the scenario? btw there is also a way to specify Avro in `components` and also have a reference to it from `payload.schema`

I basically suggest you to look on the diagram from my perspective, a doc reader. Diagrams are like text, we "read" them, component by component, and try to understand that they mean. So try to read it, and explain with text what it shows cause I do not even know what it means that `Define Avro Schema` -> `specifies`? like who specifies? What is the goal of your diagram

As example look at https://www.asyncapi.com/docs/tools/generator#generation-process where you see an example, where diagram supplements a detailed description of generation process, where text match diagram components names. Similar to https://www.asyncapi.com/docs/tools/generator/asyncapi-document#asyncapi-document-generation-process